python_modules/dagster/dagster/_core/snap/job_snapshot.py

```diff
@@ -202,12 +207,26 @@ def get_node_def_snap(self, node_def_name: str) -> Union[OpDefSnap, GraphDefSnap
 
         check.failed("not found")
 
-    def has_node_name(self, node_name: str) -> bool:
-        check.str_param(node_name, "node_name")
+    def get_retry_policy_for_node(self, node_name: str):
+        node = check.not_none(self.get_node_invocation(node_name))
+        definition = self.get_node_def_snap(node.node_def_name)
+
+        if node.retry_policy:
+            return node.retry_policy
+        elif isinstance(definition, OpDefSnap) and definition.retry_policy:
+            return definition.retry_policy
+        else:
+            return self.op_retry_policy
+
+    def get_node_invocation(self, node_name: str) -> Optional[NodeInvocationSnap]:
```

Comment:
This O(n^2) fetch if you iterate through every node name fetching retry policies may be concerning, could make a dict keys by node name